frontend/src/app/main/vote/demoday/layout.tsx

```diff
@@ -0,0 +1,47 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { usePathname } from "next/navigation"; // Next.js 13+ App Router
+import { Progress } from "@/components/ui/progress";
+import Header from "@/components/header";
+
+export default function DemodayLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  // 현재 라우트 경로
+  const pathname = usePathname();
+  // 진행도(퍼센트)
+  const [progress, setProgress] = useState(0);
+
+  useEffect(() => {
+    // pathname에 따라 진행도 결정
+    if (pathname === "/main/vote/demoday") {
+      setProgress(33);
+    } else if (pathname === "/main/vote/demoday/step2") {
+      setProgress(66);
+    } else if (pathname === "/main/vote/demoday/step3") {
+      setProgress(100);
+    } else {
+      // 그 외 경로: 0 or -1 등으로 설정
+      setProgress(0);
+    }
+  }, [pathname]);
+
+  return (
+    <div className="px-6 py-6 w-full h-full">
+        <div className="w-full flex flex-col items-center">
+            {/* Header */}
+            <Header title="데모데이 투표" className="text-logo py-6"/>
+            {/* Bar */}
+            <div className="w-full border-2 border-grey450 mb-10"></div>
+            {/* 진행 바 */}
+            <Progress value={progress} className="w-[90%] bg-grey450" />
```

Comment:
유저가 투표하는 과정에서 그 단계에 따라 진행바를 구현하신 점이 인상 깊었어요! 다만 코드를 보면 배경 바가 있고, 그 위에 진행 바가 겹쳐져서 보이는 구조같은데, 일부 환경에서는 두 요소가 완전히 겹쳐지지 않아 배경과 진행바가 살짝 따로 노는 듯한 느낌을 받았습니다...! 차라리, div안에 또다른 div를 넣는 형태로 해서 진짜 진행바가 "채워지는" 형태로 구현해보면 어떨까요?

```suggestion
<div className="w-[90%] h-4 bg-grey450 rounded-full overflow-hidden">
  <div
    className="h-full bg-blue-500 transition-all duration-300"
    style={{ width: `${progress}%` }}
  />
</div>
```